tests/unit/test_database_migration.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Unit tests for Flask charm database integration."""
+
+import ops
+import pytest
+from ops.testing import Harness
+
+from charm_state import CharmState
+from constants import FLASK_CONTAINER_NAME
+from database_migration import DatabaseMigration
+from exceptions import CharmConfigInvalidError
+from flask_app import FlaskApp
+from webserver import GunicornWebserver
+
+
+def test_database_migration(harness: Harness):
```

Comment:
Shouldn't this test be even more scoped ? It's calling `restart_flask` where I would expect unittests that covers:
`get_status`, `set_status`, `get_completed_script`, `set_completed_script`, `run` and `check_completed`

Then the webserver test modified to asserts that the corresponding methods are called with the right arguments